integration-tests/suites/threads.go

```diff
@@ -0,0 +1,62 @@
+package suites
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/stackrox/collector/integration-tests/pkg/config"
+	"github.com/stackrox/collector/integration-tests/pkg/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type ThreadsTestSuite struct {
+	IntegrationTestSuiteBase
+}
+
+func (s *ThreadsTestSuite) SetupSuite() {
+	s.RegisterCleanup("thread-exec")
+	s.StartCollector(false, nil)
+}
+
+func (s *ThreadsTestSuite) TearDownSuite() {
+	s.StopCollector()
+	s.cleanupContainers("thread-exec")
+}
+
+// Verify that Collector correctly traces threads, even if created via clone3.
+// This should lead to a correct file path, when doing exec from a thread --
+// instead of an exec target we should see the parent file path.
+func (s *ThreadsTestSuite) TestThreadExec() {
+	image := config.Images().QaImageByKey("qa-thread-exec")
+	containerID, err := s.Executor().StartContainer(
```

Comment:
I think we need to come up with a better way of handling container IDs, because it's so unintuitive whether a short or long version is required. Perhaps wrapping it in an object makes this easier so the implementations can decide which ones they need:

```
type ContainerID string

func (c *ContainerID) Long() string {
	return string(*c)
}

func (c *ContainerID) Short() string {
	return string(*c)[0:12]
}
```

Not one for this PR, of course, but for later. I might put something together.

Reply:
I think the use of `common.ContainerShortID` still needs to be added here

Reply:
Where? `ExpectProcesses` already uses it, and since the test is passing, `waitForContainerToExit` seems to be fine with the long version as well.

Reply:
ah sorry, missed that!